components/datetime/src/provider.rs

```diff
@@ -0,0 +1,232 @@
+            PatternItem::Literal(Cow::Borrowed("/")),
+            PatternItem::Field(Field {
+                symbol: FieldSymbol::Year(fields::Year::Calendar),
+                length: FieldLength::Short,
+            }),
+        ])),
+    ),
+    (
+        Skeleton(Cow::Borrowed(&[
+            Field {
+                symbol: FieldSymbol::Year(fields::Year::Calendar),
+                length: FieldLength::Short,
+            },
+            Field {
+                symbol: FieldSymbol::Month(fields::Month::Format),
+                length: FieldLength::Abbreviated,
+            },
+            Field {
+                symbol: FieldSymbol::Day(fields::Day::DayOfMonth),
+                length: FieldLength::Short,
+            },
+        ])),
+        Pattern(Cow::Borrowed(&[
+            PatternItem::Field(Field {
+                symbol: FieldSymbol::Day(fields::Day::DayOfMonth),
+                length: FieldLength::Short,
+            }),
+            PatternItem::Literal(Cow::Borrowed(" ")),
+            PatternItem::Field(Field {
+                symbol: FieldSymbol::Month(fields::Month::Format),
+                length: FieldLength::Abbreviated,
+            }),
+            PatternItem::Literal(Cow::Borrowed(" ")),
+            PatternItem::Field(Field {
+                symbol: FieldSymbol::Year(fields::Year::Calendar),
+                length: FieldLength::Short,
+            }),
+        ])),
+    ),
+    (
+        Skeleton(Cow::Borrowed(&[
+            Field {
+                symbol: FieldSymbol::Hour(fields::Hour::H23),
+                length: FieldLength::Short,
+            },
+            Field {
+                symbol: FieldSymbol::Minute,
+                length: FieldLength::Short,
+            },
+            Field {
+                symbol: FieldSymbol::Second(fields::Second::Second),
+                length: FieldLength::Short,
+            },
+        ])),
+        Pattern(Cow::Borrowed(&[
+            PatternItem::Field(Field {
+                symbol: FieldSymbol::Hour(fields::Hour::H23),
+                length: FieldLength::TwoDigit,
+            }),
+            PatternItem::Literal(Cow::Borrowed(":")),
+            PatternItem::Field(Field {
+                symbol: FieldSymbol::Minute,
+                length: FieldLength::TwoDigit,
+            }),
+            PatternItem::Literal(Cow::Borrowed(":")),
+            PatternItem::Field(Field {
+                symbol: FieldSymbol::Second(fields::Second::Second),
+                length: FieldLength::TwoDigit,
+            }),
+        ])),
+    ),
+];
+
+#[derive(Default)]
+pub struct DummyDataProvider {}
+
+impl DummyDataProvider {
+    fn get_pattern_for_style(&self, _style: &options::style::Bag) -> &Pattern<'static> {
+        let pattern = Pattern::from_bytes(b"dd/MM/y").unwrap();
+        &PATTERNS.iter().find(|(_, p)| *p == pattern).unwrap().1
+    }
+
+    pub fn get_pattern(&self, options: &DateTimeFormatOptions) -> &Pattern<'static> {
```

Comment:
I think let's just ship with dateStyle/timeStyle until we have time to figure out how to map skeletons to patterns.